sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3943,6 +3943,43 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-34421: Resolve temporary functions and views in views with CTEs") {
```

Comment:
Is this issue only related to a temp function case? How about a permanent one?

Reply:
Same question

Reply:
No, permanent functions are not affected: https://github.com/apache/spark/pull/31550#issuecomment-781229308